solutions/cses-1095.mdx

```diff
@@ -0,0 +1,59 @@
+---
+id: cses-1712
+source: CSES
+title: Exponentiation II
+author: Ryan Chou
+---
+
+## Explanation
+
+[Fermat's Little Theorem](/gold/modular) tells us that $a^{p - 1} \equiv 1 \pmod{p}$, so we can calculate $a^{b^c \pmod{1e9 + 7 - 1}} \pmod{1e9+7}$ with modular exponentiation.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(\log P)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const long long MOD = 1e9 + 7;
+
+//BeginCodeSnip{Binary Exponentiation}
+long long binpow(long long x, long long n, long long m) {
+	assert(n >= 0);
+	x %= m;
+	long long res = 1;
+	while (n > 0) {
+		if (n % 2 == 1) {  // if n is odd
+			res = res * x % m;
+		}
+		x = x * x % m;
+		n /= 2;  // divide by two
+	}
+	return res;
+}
+//EndCodeSnip
+
+int main() {
+	int t;
+	cin >> t;
+	for (int i = 0; i < t; i++) {
+		long long a;
+		long long b;
+		long long c;
+
+		cin >> a >> b >> c;
```

Comment:
```suggestion
		ll a, b, c; cin >> a >> b >> c;
```

Reply:
no?

Reply:
lmao u could write smth that took up 5 lines of code in one line

Reply:
i wrote something like this when i coded it, but changed it during formatting